falco-plugin/main.go

```diff
@@ -0,0 +1,213 @@
+		ID:                 6,
+		Name:               "seccompagent",
+		Description:        "Seccomp Agent Events",
+		Contact:            "github.com/kinvolk/seccompagent/",
+		Version:            "0.1.0",
+		RequiredAPIVersion: "0.3.0",
+		EventSource:        "seccompagent",
+	}
+}
+
+// Init is called by the Falco plugin framework as first entry,
+// we use it for setting default configuration values and mapping
+// values from `init_config` (json format for this plugin)
+func (seccompAgentPlugin *SeccompAgentPlugin) Init(config string) error {
+	seccompAgentPlugin.FlushInterval = 30
+	seccompAgentPlugin.SocketFile = "/run/seccomp-agent-falco-plugin.sock"
+	return json.Unmarshal([]byte(config), &seccompAgentPlugin)
+}
+
+// Fields exposes to Falco plugin framework all availables fields for this plugin
+func (seccompAgentPlugin *SeccompAgentPlugin) Fields() []sdk.FieldEntry {
+	return []sdk.FieldEntry{
+		{Type: "uint64", Name: "seccompagent.id", Desc: "Cookie passed by the kernel in struct seccomp_notif"},
+		{Type: "uint64", Name: "seccompagent.pid", Desc: "Process that made the syscall"},
+		{Type: "string", Name: "seccompagent.syscall", Desc: "Name of the syscall"},
+		{Type: "string", Name: "seccompagent.k8s.namespace", Desc: "Kubernetes namespace"},
+		{Type: "string", Name: "seccompagent.k8s.pod", Desc: "Kubernetes pod"},
+		{Type: "string", Name: "seccompagent.k8s.container", Desc: "Kubernetes container"},
+		{Type: "uint64", Name: "seccompagent.k8s.pid", Desc: "Pid 1 of the container"},
+		{Type: "uint64", Name: "seccompagent.k8s.pidfilter", Desc: "Process that attached the seccomp filter"},
+	}
+}
+
+// Extract allows Falco plugin framework to get values for all available fields
+func (seccompAgentPlugin *SeccompAgentPlugin) Extract(req sdk.ExtractRequest, evt sdk.EventReader) error {
+	var data SeccompAgentMessage
+
+	rawData, err := ioutil.ReadAll(evt.Reader())
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+
+	err = json.Unmarshal(rawData, &data)
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+
+	switch req.Field() {
+	case "seccompagent.id":
+		req.SetValue(data.ID)
+	case "seccompagent.pid":
+		req.SetValue(data.Pid)
+	case "seccompagent.syscall":
+		req.SetValue(data.Syscall)
+	case "seccompagent.k8s.namespace":
+		req.SetValue(data.K8SNamespace)
+	case "seccompagent.k8s.pod":
+		req.SetValue(data.K8SPod)
+	case "seccompagent.k8s.container":
+		req.SetValue(data.K8SContainer)
+	case "seccompagent.k8s.pid":
+		req.SetValue(data.K8SPid)
+	case "seccompagent.k8s.pidfilter":
+		req.SetValue(data.K8SPidFilter)
+	default:
+		return fmt.Errorf("no known field: %s", req.Field())
+	}
+
+	return nil
+}
+
+// Open is called by Falco plugin framework for opening a stream of events, we call that an instance
+func (seccompAgentPlugin *SeccompAgentPlugin) Open(params string) (source.Instance, error) {
+	ctx := context.Background()
+	msgC := make(chan SeccompAgentMessage)
+	errC := make(chan error)
```

Comment:
@alban We might want to change the channel to be buffered, so it doesn't block if one message is in the channel already.

Not sure what would be a good size, though. But we probably need to change this in a follow-up PR. What do you think?